src/js/components/AppConfigEditFormComponent.jsx

```diff
@@ -144,136 +144,178 @@ var AppConfigEditFormComponent = React.createClass({
-            </FormGroupComponent>
-          </div>
-          <div className="col-sm-3">
-            <FormGroupComponent
-              fieldId="disk"
-              label="Disk Space (MiB)"
-              errorMessage={this.getErrorMessage("disk")}
-              value={state.fields.disk}
-              onChange={this.handleFieldUpdate}>
-              <input min="0" step="any" type="number"/>
-            </FormGroupComponent>
-          </div>
-          <div className="col-sm-3">
-            <FormGroupComponent
-              fieldId="instances"
-              label="Instances"
-              errorMessage={this.getErrorMessage("instances")}
-              value={state.fields.instances}
-              onChange={this.handleFieldUpdate}>
-              <input min="0" step="1" type="number"/>
+    var containerMenuItemClassSet = classNames({
+      "error": this.fieldsHaveError({container: "container"})
+    });
+
+    var envMenuItemClassSet = classNames({
+      "error": this.fieldsHaveError({env: "env"})
+    });
+
+    var labelsMenuItemClassSet = classNames({
+      "error": this.fieldsHaveError({labels: "labels"})
+    });
+
+    var healthMenuItemClassSet = classNames({
+      "error": this.fieldsHaveError({healthChecks: "healthChecks"})
+    });
+
+    var volumesMenuItemClassSet = classNames({
+      "error": this.fieldsHaveError({localVolumes: "localVolumes"})
+    });
+
+    var optionalMenuItemClassSet = classNames({
+      "error": this.fieldsHaveError(OptionalSettingsComponent.fieldIds)
+    });
+
+    return (
+      <div className="app-config-edit">
+        <MenuComponent selected={state.activeSection} className="col-sm-3"
+            onChange={this.onMenuChange}>
+          <MenuItemComponent value="general"
+              className={generalMenuItemClassSet}>
+            General
+          </MenuItemComponent>
+          <MenuItemComponent value="container"
+              className={containerMenuItemClassSet}>
+            Docker Container
+          </MenuItemComponent>
+          <MenuItemComponent value="env"
+              className={envMenuItemClassSet}>
+            Environment Variables
+          </MenuItemComponent>
+          <MenuItemComponent value="labels"
+             className={labelsMenuItemClassSet}>
+            Labels
+          </MenuItemComponent>
+          <MenuItemComponent value="health"
+              className={healthMenuItemClassSet}>
+            Health Checks
+          </MenuItemComponent>
+          <MenuItemComponent value="volumes"
+              className={volumesMenuItemClassSet}>
+            Volumes
+          </MenuItemComponent>
+          <MenuItemComponent value="optional"
+              className={optionalMenuItemClassSet}>
+            Optional
+          </MenuItemComponent>
+        </MenuComponent>
+        <ContentComponent active={state.activeSection}
+            className="content-component">
+          <SectionComponent id="general"
```

Comment:
yeah i was tricked into believing this was an HTML id – can we please call this prop something like `sectionId` ?